lib/puppet/provider/sensuctl.rb

```diff
@@ -7,7 +8,8 @@ class Puppet::Provider::Sensuctl < Puppet::Provider
   commands :sensuctl => 'sensuctl'
 
   def config_path
-    home = File.expand_path('~')
+    user_name = Etc.getpwuid(Process.uid).name
```

Comment:
Could you please add a comment with a link to the github issue and a brief explanation about how services do not set `$HOME` which is why you cant expand `~`.

Reply:
as mentioned in #1072 when puppet is run as a daemon, ruby is not able to find the home directory and I have found the following message in the logs: 
```
//test-xxx-experiments.XXX.org//Stage[main]/Sensu::Backend/Sensu_configure[puppet] 
err, sensu_configure, puppet, class, sensu::backend, sensu, backend, xxx_experiments::sensu, xxx_experiments, role::xxx_experiments, role, node, __node_regexp__test-xxx-experiments	
Could not evaluate: couldn't find login name -- expanding `~'	/etc/puppetlabs/code/environments/test/modules/sensu/manifests/backend.pp:118
```

Reply:
So could you please put a short comment in the code?

Reply:
Got your point. Sure. Will do it tomorrow morning.